src/stan/math/prim/mat/meta/index_type.hpp

```diff
@@ -21,6 +21,16 @@ namespace stan {
       typedef typename Eigen::Matrix<T,R,C>::Index type;
     };
 
+    template <typename T>
+    struct index_type<Eigen::MatrixBase<T> > {
+      typedef typename index_type<T>::type type;
+    };
+
+    template <typename T>
+    struct index_type<Eigen::Map<T> > {
```

Comment:
Yes.  I'd suggest looking these things up!  Eigen has
great doc.  Map<T> is the expression template class that
maps contiguous memory to an Eigen type T.

- Bob

> On Mar 12, 2015, at 7:24 AM, Daniel Lee <notifications@github.com> wrote:
> 
> In src/stan/math/prim/mat/meta/index_type.hpp:
> 
> > @@ -21,6 +21,16 @@ namespace stan {
> >        typedef typename Eigen::Matrix<T,R,C>::Index type;
> >      };
> >  
> > +    template <typename T>
> > +    struct index_type<Eigen::MatrixBase<T> > {
> > +      typedef typename index_type<T>::type type;
> > +    };
> > +
> > +    template <typename T>
> > +    struct index_type<Eigen::Map<T> > {
> 
> Is Eigen::Map<T> different than Eigen::MatrixBase<T>?
> 
> —
> Reply to this email directly or view it on GitHub.
>